pkg/services/pluginsintegration/pipeline/steps.go

```diff
@@ -94,6 +94,34 @@ func (r *RegisterPluginRoles) Register(ctx context.Context, p *plugins.Plugin) (
 	return p, nil
 }
 
+// RegisterActionSets implements an InitializeFunc for registering plugin roles.
+type RegisterActionSets struct {
+	log          log.Logger
+	roleRegistry plugins.ActionSetRegistry
+}
+
+// RegisterActionSetsStep returns a new InitializeFunc for registering plugin roles.
+func RegisterActionSetsStep(actionRegistry plugins.ActionSetRegistry) initialization.InitializeFunc {
+	return newRegisterActionSets(actionRegistry).Register
+}
+
+func newRegisterActionSets(registry plugins.ActionSetRegistry) *RegisterActionSets {
+	return &RegisterActionSets{
+		log:          log.New("plugins.actionsets.registration"),
+		roleRegistry: registry,
+	}
+}
+
+// Register registers the plugin roles with the role registry.
+func (r *RegisterActionSets) Register(ctx context.Context, p *plugins.Plugin) (*plugins.Plugin, error) {
+	r.log.Info("Registering actionsets", "pluginId", p.ID, "actionSets", p.ActionSets)
+	if err := r.roleRegistry.DeclareActionSets(ctx, p.ID, p.Name, p.ActionSets); err != nil {
+		r.log.Warn("Declare plugin actionsets failed.", "pluginId", p.ID, "error", err)
```

Comment:
Was the intent to log a warning and not return the error here? Otherwise would suggest using error level

Reply:
most of the implementation is based off of roles registration. should we also log error there?

https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/pipeline/steps.go#L91